### ISSUE TITLE
init.lua:282 [proxy-cache] cache_key.lua:42: attempt to concatenate local 'arg' - Query param onyl with a name

### ISSUE THREAD

**Issue by cerias** [Kong/kong#7770]
### Summary
- Kong version 2.5.0
- proxy-cache plugin throws internal error while key building on query params without value

### Steps To Reproduce

1. Enable proxy cache without config.vary_query_param
2. Request an url with a query param, where only a name is used without a value [rfc3986](https://datatracker.ietf.org/doc/html/rfc3986#section-3.4) -> gm.woff?51482360
3. Get a internal error while the proxy cache tries to build an key to an element (which would not be cache by config but possible should be?)

### Additional Details & Logs
Error log:
`[error] 2167006#0: *82 [kong] init.lua:282 [proxy-cache] ...cal/share/lua/5.1/kong/plugins/proxy-cache/cache_key.lua:42: attempt to concatenate local 'arg' (a boolean value), client: 46.223.163.34, server: kong, request: "GET /public/theme/styles/system/fontello/font/gm.woff?51482360 HTTP/2.0", host: "host1-test20200823.xxxx"
`

Proxy-Cache config
```json 
{
  "service": null,
  "route": null,
  "id": "05b2917b-0668-4b03-b5cb-693e63f0c5de",
  "config": {
    "cache_control": true,
    "strategy": "memory",
    "memory": {
      "dictionary_name": "kong_db_cache"
    },
    "response_code": [
      200,
      301,
      404
    ],
    "storage_ttl": null,
    "vary_query_params": null,
    "content_type": [
      "\"text/css\""
    ],
    "vary_headers": null,
    "request_method": [
      "GET",
      "HEAD"
    ],
    "cache_ttl": 300
  },
  "consumer": null,
  "protocols": [
    "grpc",
    "grpcs",
    "http",
    "https"
  ],
  "enabled": true,
  "name": "proxy-cache",
  "tags": null,
  "created_at": 1629980813
}```


If you need more details or informations i will be happy to help.

**Comment by ErrorProne** [comment]
I can second this. Given that I've no experience at all with lua and/or kong and just stumbled upon this, the following seems to fix the problem in my test env in cache_key.lua:42
`        insert(cache_key, field .. "=" .. tostring(arg))`

However I just wanted to open a pull request and have seen that this has already been fixed on the master branch 11 month ago: https://github.com/Kong/kong/blob/master/kong/plugins/proxy-cache/cache_key.lua#L46

Which is not at all what I got when installing the 2.5.0 version via the ubuntu kong package https://download.konghq.com/gateway-2.x-ubuntu-focal/. While checking the 2.5.0 artifact on github I found that there is now proxy-cache plugin at all, which leaves me a little bit puzzled about the bundling and deployment process. Could it be that there is a problem in the bundle?

**Comment by locao** [comment]
Hello @cerias and @ErrorProne! Thanks for the report and sorry for the delayed answer. 

Yes, @ErrorProne is right: this issue was already fixed. The fix was merged in #7738 and was released with Kong 2.6.0.

I'm closing this issue as it was already addressed. Please feel free to reopen it if you find the same issue again.
